tests/update/update_workflow_sdk_test.go

```diff
@@ -90,69 +78,118 @@ func (s *UpdateWorkflowClientSuite) TestUpdateWorkflow_TerminateWorkflowAfterUpd
 	var notFound *serviceerror.NotFound
 	s.ErrorAs(err, &notFound)
+
+	s.EqualHistoryEvents(`
+  1 WorkflowExecutionStarted
+  2 WorkflowTaskScheduled
+  3 WorkflowTaskStarted
+  4 WorkflowTaskFailed
+  5 WorkflowExecutionTerminated`, s.GetHistory(s.Namespace(), tv.WorkflowExecution()))
 }
 
-// TestUpdateWorkflow_TerminateWorkflowDuringUpdate executes a long-running update (schedules a sequence of activity
-// calls) and terminates the workflow after the update has been accepted but before it has been completed. It checks
+// TestUpdateWorkflow_TimeoutWorkflowAfterUpdateAccepted executes an update, and while WF awaits
+// server times out the WF after the update has been accepted but before it has been completed. It checks
 // that the client gets a NotFound error when attempting to fetch the update result (rather than a timeout).
-func (s *UpdateWorkflowClientSuite) TestUpdateWorkflow_TerminateWorkflowAfterUpdateAccepted() {
+func (s *UpdateWorkflowSdkSuite) TestUpdateWorkflow_TimeoutWorkflowAfterUpdateAccepted() {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 	defer cancel()
 	tv := testvars.New(s.T()).WithTaskQueue(s.TaskQueue()).WithNamespaceName(namespace.Name(s.Namespace()))
 
-	activityDone := make(chan struct{})
-	activityFn := func(ctx context.Context) error {
-		activityDone <- struct{}{}
-		return nil
+	workflowFn := func(ctx workflow.Context) error {
+		s.NoError(workflow.SetUpdateHandler(ctx, tv.HandlerName(), func(ctx workflow.Context, arg string) error {
+			ctx = workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+				StartToCloseTimeout: 10 * time.Second,
+			})
+			s.NoError(workflow.Await(ctx, func() bool { return false }))
+			return unreachableErr
+		}))
+		s.NoError(workflow.Await(ctx, func() bool { return false }))
+		return unreachableErr
 	}
 
+	s.Worker().RegisterWorkflow(workflowFn)
+
+	wfRun, err := s.SdkClient().ExecuteWorkflow(ctx, sdkclient.StartWorkflowOptions{
+		ID:                       tv.WorkflowID(),
+		TaskQueue:                tv.TaskQueue().Name,
+		WorkflowExecutionTimeout: time.Second,
+	}, workflowFn)
+	s.NoError(err)
+
+	updateHandle, err := s.updateWorkflowWaitAccepted(ctx, tv, "my-update-arg")
+	s.NoError(err)
+
+	var notFound *serviceerror.NotFound
+	s.ErrorAs(updateHandle.Get(ctx, nil), &notFound)
+
+	_, pollErr := s.pollUpdate(ctx, tv, &updatepb.WaitPolicy{LifecycleStage: enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED})
+	s.ErrorAs(pollErr, &notFound)
+
+	var wee *temporal.WorkflowExecutionError
+	s.ErrorAs(wfRun.Get(ctx, nil), &wee)
+
+	s.EqualHistoryEvents(`
+  1 WorkflowExecutionStarted
+  2 WorkflowTaskScheduled
+  3 WorkflowTaskStarted
+  4 WorkflowTaskCompleted
+  5 WorkflowExecutionUpdateAccepted
+  6 WorkflowExecutionTimedOut`, s.GetHistory(s.Namespace(), tv.WorkflowExecution()))
+}
+
+// TestUpdateWorkflow_TerminateWorkflowDuringUpdate executes an update, and while WF awaits
```

Comment:
Name in docstring doesn't match function.

Reply:
![](https://user-images.githubusercontent.com/2232524/75076254-832f8f00-54b4-11ea-81b9-1a6648a486a5.png)